components/microsoft_onedrive/actions/Upload-file

```diff
@@ -0,0 +1,58 @@
+import fs from "fs";
+import onedrive from "../../microsoft_onedrive.app.mjs";
+import httpRequest from "../../common/httpRequest.mjs";
+import { ConfigurationError } from "@pipedream/platform";
+
+export default {
+  name: "Upload File",
+  description: "Upload a file to OneDrive.",
+  key: "microsoft_onedrive-upload-file",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    onedrive,
+    uploadFolderId: {
+      type: "string",
+      label: "Upload Folder ID",
+      description: "The ID of the folder where you want to upload the file.",
+      optional: false,
+    },
```

Comment:
Here you must to add the async options to list the folders. You can see an example [here](https://github.com/PipedreamHQ/pipedream/blob/33d11f0cb100c7c29a30f45f860abce7dca95481/components/github/github.app.mjs#L55)